sunpy/map/mapbase.py

```diff
@@ -2258,7 +2258,7 @@ def plot(self, annotate=True, axes=None, title=True,
         if not axes:
             axes = wcsaxes_compat.gca_wcs(self.wcs)
 
-        if not wcsaxes_compat.is_wcsaxes(axes):
+        if not wcsaxes_compat.is_same_wcsaxes(axes, self.wcs):
             warnings.warn("WCSAxes not being used as the axes object for this plot."
```

Comment:
This feels like the wrong warning for a mismatch? Also the mismatch should probably be an error in 3.0.